src/TheSpace/HarbergerMarket.sol

```diff
@@ -103,12 +104,20 @@ contract HarbergerMarket is ERC721, IHarbergerMarket, Multicall, AccessRoles {
         public
         view
         virtual
-        override(AccessControl, ERC721, IERC165)
+        override(AccessControl, ERC721, IERC165, ERC721Enumerable)
         returns (bool)
     {
         return super.supportsInterface(interfaceId_);
     }
 
+    function _beforeTokenTransfer(
```

Comment:
If we replace ERC721 with ERC721Enumerable, we can remove this as well.

Reply:
these changes have done.